backend/src/schema/repository.ts

```diff
@@ -80,6 +81,10 @@ const resolvers = {
       }
 
       await saveChanges(saveArgs, context.currentUser)
+      .catch((error: GitError) => {
+        console.log(error)
+        throw new Error('Merge conflict detected')
+      })
```

Comment:
This probably throws this "merge error" for all git errors in `saveChanges` (such as config errors or something). Have you thought about this? Would error checking in `backend/src/services/git.ts/gitCommit` function be possible with checking if merge is possible before committing (i.e. with something like `git merge --no-commit --no-ff BRANCH-TO-MERGE`) and acting according to that output?

Reply:
Yeah, that's true. I'll check that later.

Reply:
Okay. Btw, SimpleGit offers a way to run any git command with the `.raw` method. Example commit operation could just be done with `await git.raw(['commit', '-m', 'commitmessage'])`. This might be useful when you want to run any special commands.

Reply:
Couldn't check if merge is possible with `await git.merge(['--no-commit', '--no-ff', branch]).result`, it just showed `success`.

Reply:
Thought about this more and what do you think about the following flow of this error:

1. Do the local commit
2. Before pushing do the following:
    1. `git fetch` which fetches remote changes for this branch
    2. `git merge origin/branchname` which merges your changes
        * if this auto-merge catches error and error message contains "CONFLICT" then do `git merge --abort` and throw merge conflict error. Other error -> "Unexpected error" for some other case
        * else just push the merged stuff


Here's an example code that I tested briefly. The idea is that of course we can't know if a merge conflict happens without checking the merge to remote. Tell me what you think and others too

```
const gitPush = async (
  git: SimpleGit,
  username: string,
  token: string,
  branchName: string
) => {
  await git.fetch()
  await git.merge([`origin/${branchName}`]).catch(async (error: GitError) => {
    await git.merge(['--abort'])
    if (error.message.includes('CONFLICT')) {
      throw new Error('merge conflict')
    }
    throw new Error('Unexpected error')
  })

  const remoteUuid = uuidv4()
  await gitAddRemote(git, remoteUuid, username, token)
  await git.push(remoteUuid, branchName)
  await gitRemoveRemote(git, remoteUuid)
}
```

Reply:
Example code seems to lead to error in cloning repo after it fails but this was just to clarify my idea how a merge conflict might be detected.

Reply:
This looks useful, but should we include this already in this branch or create a whole new issue about error handling? The generated "loose" commit should be handled someway + more cases should be checked.

Reply:
Perhaps just do `git reset --hard HEAD~1` after merge conflict is detected? It would remove the commit just made. 

Example: 

`
  } catch (e) {
    await git.merge(['--abort'])
    await git.reset(['--hard', 'HEAD~1']) (or await git.raw(['reset','--hard','HEAD~1']) whichever works) 
    throw e
  }
`

Reply:
Now these suggestions are used. There were more errors when saving to a new branch after conflict. Saving to a new branch didn't succeed if the branch name didn't exist because git complained `GitError: merge: origin/<branchName> - not something we can merge`. Ignored that in error handling.